src/wifi.c

```diff
@@ -0,0 +1,84 @@
+#include <freertos/FreeRTOS.h>
+#include <freertos/event_groups.h>
+#include <esp_event_loop.h>
+#include <esp_wifi.h>
+#include <string.h>
+#include <stdlib.h>
+#include <nvs_flash.h>
+
+#include "wifi.h"
+
+#define CONNECTED_BIT BIT0
+
+static EventGroupHandle_t wifi_event_group;
+
+/**
+ * Handler for WiFi events
+ *
+ * @ctx: context pointer
+ * @event: Wifi event that occurred
+ *
+ * @return: esp_err_t 0 in every case
```

Comment:
why resolve the ESP_OK macro here?

Reply:
This is the current state. If we want to add different return values later, we can do so.

Reply:
this doesn't answer the question why resolve the macro

Reply:
No, but I think it's easier to understand what is returned with a value, not a macro. While ESP_OK might be obvious, in documentation i see no point in hiding return values behind macros

Reply:
okay, but this implies knowledge of the implementation to deduct semantic,
which is a pretty bad pattern imho. naming the type as `esp_err_t` is not the problem but introducing magic values is.

Reply:
especially documentation should no be implementation dependent right?

Reply:
This does not introduce a magic value, but describes explicitly what is returned

Reply:
it describes the actual value not the semantic value, but ok.